stream-chat-android-offline/src/main/java/io/getstream/chat/android/livedata/EventHandlerImpl.kt

```diff
@@ -310,22 +310,43 @@ internal class EventHandlerImpl(
 
                 is MessageReadEvent -> {
                     // get the channel, update reads, write the channel
-                    batch.getCurrentChannel(event.cid)?.let {
-                        val updatedChannel = it.apply {
-                            updateReads(ChannelUserRead(user = event.user, lastRead = event.createdAt))
+                    event.cid
+                        ?.let(batch::getCurrentChannel)
+                        ?.apply {
+                            updateReads(
+                                ChannelUserRead(
+                                    user = event.user,
+                                    lastRead = event.createdAt
+                                )
+                            )
                         }
-                        batch.addChannel(updatedChannel)
-                    }
+                        ?.let(batch::addChannel)
+                        ?: batch
+                            .getAllChannels()
+                            .forEach { channel ->
+                                channel.updateReads(ChannelUserRead(event.user, event.createdAt))
```

Comment:
Why do we update all all channels?

Reply:
this is not good, we should implement this with 1 write, see Jira